rust/datafusion/src/physical_plan/expressions.rs

```diff
@@ -2604,6 +2678,61 @@ mod tests {
         )
     }
 
+    #[test]
+    #[rustfmt::skip]
+    fn nullif_int32() -> Result<()> {
+        let schema = Schema::new(vec![Field::new("a", DataType::Int32, false)]);
+        let a = Int32Array::from(vec![Some(1), Some(2), None, None, Some(3), None, None, Some(4), Some(5)]);
+        let a = Arc::new(a);
+        let a_len = a.len();
+        let batch = RecordBatch::try_new(Arc::new(schema.clone()), vec![a.clone()])?;
+
+        let literal_expr = lit(ScalarValue::Int32(Some(2)));
+        let lit_array = literal_expr.evaluate(&batch)?;
+
+        let result = nullif_func(&[a.clone(), lit_array])?;
+
+        // Results should be: Some(1), None, None, None, Some(3), None
+        assert_eq!(result.len(), a_len);
+
+        let result = result
+            .as_any()
+            .downcast_ref::<Int32Array>()
+            .expect("failed to downcast to Int32Array");
+
+        assert_eq!(1, result.value(0));
+        assert_eq!(true, result.is_null(1));    // 2==2, slot 1 turned into null
+        assert_eq!(true, result.is_null(2));
+        assert_eq!(true, result.is_null(3));
+        assert_eq!(3, result.value(4));
+        assert_eq!(true, result.is_null(5));
+        assert_eq!(true, result.is_null(6));
+        assert_eq!(5, result.value(8));
+        Ok(())
```

Comment:
Same here: easier to debug if there is a single `assert_eq`, as we can see the whole array in the message.

Reply:
👍